src/opus.c

```diff
@@ -135,6 +147,17 @@ OPUS_EXPORT void opus_pcm_soft_clip(float *_x, int N, int C, float *declip_mem)
       declip_mem[c] = a;
    }
 }
+
+void opus_pcm_soft_clip_with_arch(float *_x, int N, int C, float *declip_mem, int arch)
```

Comment:
This doesn't seem very useful. You might as well just call the impl() function from everywhere, instead of the _with_arch() one that then calls impl().